components/builder-originsrv/src/migrations/origin_packages.rs

```diff
@@ -42,44 +42,94 @@ pub fn migrate(migrator: &mut Migrator) -> Result<()> {
              )"#,
     )?;
     migrator.migrate("originsrv",
-                 r#"CREATE OR REPLACE FUNCTION insert_origin_package_v1 (
-                    op_origin_id bigint,
-                    op_owner_id bigint,
-                    op_name text,
-                    op_ident text,
-                    op_checksum text,
-                    op_manifest text,
-                    op_config text,
-                    op_target text,
-                    op_deps text,
-                    op_tdeps text,
-                    op_exposes text
-                 ) RETURNS SETOF origin_packages AS $$
-                     DECLARE
-                        inserted_package origin_packages;
-                        channel_id bigint;
-                     BEGIN
-                         INSERT INTO origin_packages (origin_id, owner_id, name, ident, checksum, manifest, config, target, deps, tdeps, exposes)
-                                VALUES (op_origin_id, op_owner_id, op_name, op_ident, op_checksum, op_manifest, op_config, op_target, op_deps, op_tdeps, op_exposes)
-                                RETURNING * into inserted_package;
+                 r#"CREATE OR REPLACE FUNCTION insert_origin_package_v2 (
```

Comment:
I have a mild concern that we need to change insert_origin_package.  I guess we have to do it for consistency of the return value between insert_origin_package and get_origin_package.  I guess the high level design question is, are we exposing channels everywhere origin packages are consumed, or only for specific cases.  If everywhere, then not clear why we need a specific api like 'list_origin_packages_for_version'.  If only for specific cases, then why do we need the core functions like insert/get for origin packages modified?